### ISSUE TITLE
ST Cloudless Fdroid, Nougat: File creation/saving issues

### ISSUE THREAD

**Issue by ageru** [mpcjanssen/simpletask-android#861]
Version: ST Nextcloud (Cloudless) Fdroid 10.0.7
Phone: F5321, Android 7.1.1 (Nougat)

Steps to reproduce:
1) Install the app from F-droid
2) Grant the app Storage access

Actual result: 
- at the root of the internal memory or SD card, **/data**/nl.mpcjanssen.simpletask/ is created
- This folder is not recreated if I delete it, uninstall the app and reinstall it through f-droid.
- It's impossible to save any todo.txt in any folder, unless you recreate nl.mpcjanssen.simpletask/ in **/Android/data**/ or **/Android/media**/

Expected result
- nl.mpcjanssen.simpletask/ should be created in the proper spot, either in **/Android/data**/ or **/Android/media/**, at the root of user accessible internal or sdcard storage. The Android/media folder looks like the best spot, but I'll admit I haven't researched this.
- I am uncertain about this, but it seems granting the app Storage access should allow to save the file pretty much anywhere else in user accessible storage

Edit: emphasis added on paths, to avoid confusion

**Comment by smichel17** [comment]
Related #722. I thought I'd linked the relevant documentation from there, but apparently not, let me find that..

**Comment by mpcjanssen** [comment]
Creating a todo.txt file by other means somewhere on internal storage should allow you to open and use that.  don't know why the data folder is not recreated (even on re-install) that's very odd.

**Comment by ageru** [comment]
Thanks for answering!

Unfortunately, I was importing an existing todo.txt, and I couldn't edit it if it was anywhere but in /Android/data/nl.mpcjanssen.simpletask/ or /Android/media/nl.mpcjanssen.simpletask/.

I have upgraded my phone to Oreo since. As soon as F-droid carries a version fixing the filter issues (10.1.3 is not cutting it for me, had to revert to 10.0.7), I'll conduct more tests.

**Comment by ageru** [comment]
So, I moved to a new phone...

Version: ST Nextcloud (Cloudless) Fdroid 10.1.13
Phone: G8342, Android 8.0 (Oreo)
No SD card added, only internal storage used.

I could not reproduce the bug.
I was able to use a todo.txt synchronised from nextcloud and update it both ways.

So yeah, seems my previous phone had a brain fart. I apologies for the noise.

Just in case, the only configuration difference I could find between the 2 phones was in Other > root://directory.

- Old phone : `/storage/emulated/0/Android/media/nl.mpcjanssen.simpletask`
- New phone: `/sdcard/`

Not sure what this option does, but I did fiddle with it to try to solve my issue. Maybe that was part of the problem, maybe not, but I'll leave it here in case it's useful to someone.
